src/kiwixapp.cpp

```diff
@@ -11,31 +11,25 @@
 #include <QPrinter>
 #include <QPrintDialog>
 #include <thread>
-
-kiwix::Downloader* createDownloader() {
-    int attempt = 5;
-    while(attempt--) {
-        try {
-            return new kiwix::Downloader();
-        } catch (exception& e) {
-            qInfo() << "Cannot create downloader" << e.what();
-            std::this_thread::sleep_for(std::chrono::milliseconds(500));
-        }
-    }
-    return nullptr;
-}
+#include <QMessageBox>
 
 KiwixApp::KiwixApp(int& argc, char *argv[])
     : QApplication(argc, argv),
       m_settingsManager(),
       m_libraryDirectory(findLibraryDirectory()),
       m_library(),
-      mp_downloader(createDownloader()),
-      m_manager(&m_library, mp_downloader),
+      mp_downloader(nullptr),
       mp_server(new kiwix::KiwixServe(
         appendToDirectory(m_libraryDirectory.toStdString(),"library.xml"),
         m_settingsManager.getKiwixServerPort()))
 {
+    try {
+        mp_downloader = new kiwix::Downloader();
+    } catch (exception& e) {
+        QMessageBox::critical(nullptr, tr("Cannot create downloader"),
+        tr("Impossible to launch downloader, Kiwix-desktop will start but all download functions will not working !<br><br>") + e.what());
```

Comment:
`tr` will try to load the localized text. So we should install the translators before.

Reply:
We will not use the tr() fct to translate, I want to remove all the "hard coded" text in another PR to fix https://github.com/kiwix/kiwix-desktop/issues/299 and use a translate file instead to work with translateWiki. This solution should also make it possible to better manage translations in HTML/JS files such as the contentmanager page.

Reply:
Ok, I don't know how translateWiki works. But it will probably use a function to get the text
to display to the user. And this function must be used after the configuration of the
translation system.

And for now, we are using `tr`. We must move this after the installation of the translators
anyway. At least to have the translation working while we are moving to translateWiki.